src/calendar-app/workerUtils/worker/CalendarWorkerLocator.ts

```diff
@@ -0,0 +1,473 @@
+	cache: EntityRestInterface
+	cachingEntityClient: EntityClient
+	eventBusClient: EventBusClient
+	rsa: RsaImplementation
+	kyberFacade: KyberFacade
+	pqFacade: PQFacade
+	entropyFacade: EntropyFacade
+	blobAccessToken: BlobAccessTokenFacade
+	keyCache: KeyCache
+	keyLoader: KeyLoaderFacade
+	keyRotation: KeyRotationFacade
+
+	// login
+	user: UserFacade
+	login: LoginFacade
+
+	// domains
+	blob: lazyAsync<BlobFacade>
+	mail: lazyAsync<MailFacade>
+	calendar: lazyAsync<CalendarFacade>
+	counters: lazyAsync<CounterFacade>
+	Const: Record<string, any>
+
+	// management facades
+	groupManagement: lazyAsync<GroupManagementFacade>
+	userManagement: lazyAsync<UserManagementFacade>
+	recoverCode: lazyAsync<RecoverCodeFacade>
+	customer: lazyAsync<CustomerFacade>
+	giftCards: lazyAsync<GiftCardFacade>
+	mailAddress: lazyAsync<MailAddressFacade>
+	booking: lazyAsync<BookingFacade>
+	share: lazyAsync<ShareFacade>
+	cacheManagement: lazyAsync<CacheManagementFacade>
+
+	// misc & native
+	configFacade: lazyAsync<ConfigurationDatabase>
+	deviceEncryptionFacade: DeviceEncryptionFacade
+	native: NativeInterface
+	workerFacade: WorkerFacade
+	sqlCipherFacade: SqlCipherFacade
+	pdfWriter: lazyAsync<PdfWriter>
+
+	// used to cache between resets
+	_worker: CalendarWorkerImpl
+	_browserData: BrowserData
+
+	//contact
+	contactFacade: lazyAsync<ContactFacade>
+}
+export const locator: CalendarWorkerLocatorType = {} as any
+
+export async function initLocator(worker: CalendarWorkerImpl, browserData: BrowserData) {
+	locator._worker = worker
+	locator._browserData = browserData
+	locator.keyCache = new KeyCache()
+	locator.user = new UserFacade(locator.keyCache)
+	locator.workerFacade = new WorkerFacade()
+	const dateProvider = new NoZoneDateProvider()
+
+	const mainInterface = worker.getMainInterface()
+
+	const suspensionHandler = new SuspensionHandler(mainInterface.infoMessageHandler, self)
+	locator.instanceMapper = new InstanceMapper()
+	locator.rsa = await createRsaImplementation(worker)
+
+	const domainConfig = new DomainConfigProvider().getCurrentDomainConfig()
+
+	locator.restClient = new RestClient(suspensionHandler, domainConfig)
+	locator.serviceExecutor = new ServiceExecutor(locator.restClient, locator.user, locator.instanceMapper, () => locator.crypto)
+	locator.entropyFacade = new EntropyFacade(locator.user, locator.serviceExecutor, random, () => locator.keyLoader)
+	locator.blobAccessToken = new BlobAccessTokenFacade(locator.serviceExecutor, dateProvider, locator.user)
+	const entityRestClient = new EntityRestClient(locator.user, locator.restClient, () => locator.crypto, locator.instanceMapper, locator.blobAccessToken)
+
+	locator.native = worker
+	locator.booking = lazyMemoized(async () => {
+		const { BookingFacade } = await import("../../../common/api/worker/facades/lazy/BookingFacade.js")
+		return new BookingFacade(locator.serviceExecutor)
+	})
+
+	let offlineStorageProvider
+	if (isOfflineStorageAvailable() && !isAdminClient()) {
```

Comment:
Do we need the `isAdminClient` check in the calendar? It shouldn't ever be admin client, right?

Reply:
yeah, we don't need admin client stuff in here